tests/model_serving/model_server/multi_node/test_nvidia_multi_node.py

```diff
@@ -62,12 +68,161 @@ def test_multi_node_pods_distribution(self, multi_node_predictor_pods_scope_clas
 
         assert pods_nodes.issubset({node.name for node in nvidia_gpu_nodes}), "Pods not running on GPU nodes"
 
-    def test_multi_node_basic_inference(self, multi_node_inference_service):
-        """Test multi node basic inference"""
+    def test_multi_node_basic_internal_inference(self, multi_node_inference_service):
+        """Test multi node basic internal inference"""
         verify_inference_response(
             inference_service=multi_node_inference_service,
             inference_config=VLLM_INFERENCE_CONFIG,
             inference_type="completions",
             protocol=Protocols.HTTP,
             use_default_query=True,
         )
+
+    @pytest.mark.tls
+    def test_tls_secret_exists_in_control_ns(self, multi_node_inference_service, ray_ca_tls_secret):
+        """Test multi node ray ca tls secret exists in control (applications) namespace"""
+        if not ray_ca_tls_secret.exists:
+            raise ResourceNotFoundError(
+                f"Secret {ray_ca_tls_secret.name} does not exist in {ray_ca_tls_secret.namespace} namespace"
+            )
+
+    @pytest.mark.tls
+    def test_tls_secret_exists_in_inference_ns(self, ray_tls_secret):
+        """Test multi node ray tls secret exists in isvc namespace"""
+        if not ray_tls_secret.exists:
+            raise ResourceNotFoundError(f"Secret {ray_tls_secret.name} does not exist")
+
+    @pytest.mark.tls
+    def test_cert_files_exist_in_pods(self, multi_node_predictor_pods_scope_class):
+        """Test multi node cert files exist in pods"""
+        missing_certs_pods = []
+        for pod in multi_node_predictor_pods_scope_class:
+            certs = pod.execute(command=["ls", "/etc/ray/tls"]).split()
+            if "ca.crt" not in certs or "tls.pem" not in certs:
+                missing_certs_pods.append(pod.name)
+
+        assert not missing_certs_pods, f"Missing certs in pods: {missing_certs_pods}"
```

Comment:
I think assert in test is not recommended. If run with -o or -oo option asserts are not caught. Instead using `if` condition is recommended.

Reply:
done